week2/project/ecommerce/src/components/CategoryList.jsx

```diff
@@ -0,0 +1,63 @@
+import Category from "./Category";
+
+import "./components-css/category.css";
+import { useEffect, useState } from "react";
+import axios from "axios";
+
+function CategoryList(props) {
+  const [categoriesData, setCategoriesData] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    getCategoriesData();
+  }, []);
+
+  async function getCategoriesData() {
+    await axios("https://fakestoreapi.com/products/categories")
+      .then((respose) => {
+        setCategoriesData(respose.data);
+      })
+      .catch((error) => {
+        console.error("Error fetching data: ", error);
+        setError(error);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }
+
+  if (loading) return "Loading ...";
+  if (error) return "Error!";
```

Comment:
would be great if  you return components for loading spinner or showing the error
sure in future no need for homework